sdks/python/Makefile

```diff
@@ -1,18 +1,16 @@
-GIT_TAG               := $(shell git describe --exact-match --tags --abbrev=0 2> /dev/null || echo untagged)
-ifeq ($(GIT_TAG),untagged)
-VERSION               := 0.0.0-latest
-else
-# remove the "v" prefix, not allowed
-VERSION               := $(subst v,,$(GIT_TAG))
-endif
+# SDK version
+VERSION := $(shell ./sdk_version.py)
 
 # work dir
-WD                    := $(shell echo "`pwd`/client")
+WD := $(shell echo "`pwd`/client")
 
 DOCKER = docker run --rm -v $(WD):/wd --workdir /wd
 CHOWN = chown -R $(shell id -u):$(shell id -g)
 
-publish: generate
+verify_version:
+	@if [ ${VERSION} == "FAILED" ]; then echo "Failed to obtain a valid version from the SDK versioning script! Version: ${VERSION}"; exit 1; fi
```

Comment:
This "FAILED" might not be needed. We can use `$?` to obtain the exit status of the python script (any exceptions will be non-zero exit). We can also make a big try-catch block and `os.exit(1)` expliclty during exceptions.

Reply:
Ended up using a `ifndef` block.